### ISSUE TITLE
Possible bug in Node._findSchema

### ISSUE THREAD

**Issue by dfnjy** [josdejong/jsoneditor#961]
Hey I was having issue with a nested enum field not showing up as a select box. My setup is like follows:
```
Anyof:[
{...enum somewhere},
{...another enum somewhere}
]
```
I checked out the package and looked into the code, I guess Node._findSchema won't work properly. At the end of that method an if condition says "if path.length >0 then return null". But if the path parameter passed in has a length greater than 0, there's no line of code that will change it to 0. So the logic will be:
1. I'm looking for a schema match with path > 0
2. I found the schema match by dfs, deep into the search tree somewhere
3. After returning to the current level, the path is still > 0, so I return null despite of I found the match previously

Please correct me if I'm wrong and let me know if I can give more details.
(I'm not sure about the correct way to fix that method so no PR associated)

Thanks!

**Comment by josdejong** [comment]
Thanks for reporting @dfnjy . Can you share a (preferably small) example JSON schema and document to reproduce the issue?

**Comment by dfnjy** [comment]
Here you go:
```
<script>
  const schema02 = {
    "title": "Product name 1",
    "description": "A product in the catalog",
    "type": "object",
    "required": ["firstName", "lastName"],
    "properties": {
      "firstName": {
        "type": "string",
        "pattern": "schema02"
      },
      "lastName": {
        "description": "Id of the product",
        "type": "string",
        "enum": ["a","b","c"]
      }
    },
    "additionalProperties": false
  };
  const schema03 = {
    "title": "Product name 2",
    "description": "A product in the catalog",
    "type": "object",
    "required": ["firstName", "lastName"],
    "properties": {
      "firstName": {
        "type": "string",
        "pattern": "schema03"
      },
      "lastName": {
        "description": "Id of the product",
        "type": "string",
        "enum": ["d","e","f"]
      }
    },
    "additionalProperties": false
  };
  const schemaAnyOf = {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "http://example.com/product.schema.json",
    "title": "Product",
    "description": "A product in the catalog",
    "type": "object",
    "required": ["productId", "productName1", "productName2"],
    "properties": {
      "productId": {
        "type": "string"
      },
      "productName1": {
        "anyOf": [schema02, schema03]
      },
      "productName2": {
        "anyOf": [schema02, schema03]
      }
    },
    "additionalProperties": false
  };
  const schemaIfElse = {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "http://example.com/product.schema.json",
    "title": "Product",
    "description": "A product in the catalog",
    "type": "object",
    "required": ["productId", "productName1", "productName2"],
    "properties": {
      "productId": {
        "type": "string"
      },
      "productName1": {
        "if": { "properties": { "productId": { "pattern": '^aa$' } } },
        "then": schema02,
        "else": schema03
      },
      "productName2": {
        "if": { "properties": { "productId": { "pattern": '^bb$' } } },
        "then": schema02,
        "else": schema03
      }
    },
    "additionalProperties": false
  };
  const templates = [
    {
      text: 'Product Name',
      title: 'Insert a Product Node',
      className: 'jsoneditor-type-object',
      field: 'productName',
      value: {
        'firstName': '',
        'lastName': ''
      }
    }
  ];
  // create the editor
  const container = document.getElementById('jsoneditor');
  const options = {
    schema: schemaAnyOf,
    templates
  };
  const editor = new JSONEditor(container, options);

  // set json
  document.getElementById('setJSON').onclick = function () {
    const json = {
      'productId': '',
      'productName': {
        'firstName': '',
        'lastName': ''
      }
    }
    editor.set(json)
  }

  // get json
  document.getElementById('getJSON').onclick = function () {
    const json = editor.get()
    alert(JSON.stringify(json, null, 2))
  }
</script>
```
Step to reproduce:
1. Run this example. (I found I cannot open the context menu to insert when the body is empty, the initial state. Is it happening on your side too?)
2. Insert two fields, `productName1` and `productName2`, by using the template.
3. Verify that both fields have "lastName" rendered as dropdown.

Some additional questions:
1. I also include an if else schema too. That's because the anyOf will give you some misleading errors. Like if I have a `productName1.firstName = schema02`, then I would expect the only error message to be `productName1.lastName can only be one of [a,b,c]`. But actually I will see the firstName should be "schema03" and/or a dropdown (or not) to be [d,e,f]...
2. Is if else schame supported? I didn't find any related code actually...
3. May I know the estimated time of fixing this, and time of supporting if/else? These features are kind of important to me, and if it will take several weeks I may consider reinventing the wheel.

Thank you!

**Comment by josdejong** [comment]
Thanks, I can reproduce your issue 👍 

It looks like the enum dropdown is not created when JSON is inserted via the template. When I change the property `lastName` (which should have the enum `["d","e","f"]`) to `lastNameFoo` and then change it back to `lastName`, the enum dropdown is created.

> Run this example. (I found I cannot open the context menu to insert when the body is empty, the initial state. Is it happening on your side too?)

This is a bug introduced a few days ago in `v8.6.5`, see #969, will be fixed soon.

**Comment by josdejong** [comment]
I see I forgot to close this issue, it should be solved in `v8.6.6` published two months ago.
